File: met2.py
import random
i=0
stringlist=["ekfrasi"]
count=1
while i<=len(stringlist) :
	if stringlist[i]=="ekfrasi":
		print("Step",count,":"," ".join(stringlist))
		x=random.randint(0,1)
		if x==0:
			stringlist.remove("ekfrasi")
			stringlist.insert(i,"oros")
		else :
			stringlist.insert(i+1,"+")
			stringlist.insert(i+2,"oros")
	elif stringlist[i]=="oros":
		print("Step",count,":"," ".join(stringlist))
		x=random.randint(0,1)
		if x==0 :
			stringlist.remove("oros")
			stringlist.insert(i,"paragontas")
		else :
			stringlist.insert(i+1, "*")
			stringlist.insert(i+2, "paragontas")
	elif stringlist[i]=="paragontas":
		print("Step",count,":"," ".join(stringlist))
		x=random.randint(0,2)
		if x==0:
			stringlist.remove("paragontas")
			stringlist.insert(i,"a")
		elif x==1:
			stringlist.remove("paragontas")
			stringlist.insert(i,"b")
		else:
			stringlist.remove("paragontas")
			stringlist.insert(i,"c")
	else:
		i+=2
	count+=1
print("Generated String:"," ".join(stringlist))

	

File: met2.cpp
#include <iostream>
#include <string>
#include <stdlib.h>
#include <time.h>

using namespace std;

int main(void){
	
	string str1 = "ekfrasi";
	int i=0,count=1;
	srand (time(NULL));
	
	while(i<=str1.length()){
		
		if(str1.substr(i,7)=="ekfrasi"){
			cout<<"Step "<<count<<": "<<str1<<endl;
			int x =rand() % 10;
			if ((x%2)==1){
				str1.replace(i,7,"oros");
			}
			else{
				str1.insert(i+7,"+oros");
			}
		}
		else if(str1.substr(i,4)=="oros"){
			cout<<"Step "<<count<<": "<<str1<<endl;
			int x =rand() % 10 ;
			if ((x%2)==1){
				str1.replace(i,4,"paragontas");
			}
			else{
				str1.insert(i+4,"*paragontas");
			}
		}
		else if(str1.substr(i,10)=="paragontas"){
			cout<<"Step "<<count<<": "<<str1<<endl;
			int x =rand() % 3 + 1;
			if (x==1){
				str1.replace(i,10,"a");
			}
			else if (x==2){
				str1.replace(i,10,"b");
			}
			else{
				str1.replace(i,10,"c");
			}
		}
		else{
			i+=1;
			count--;
		}
		count++;
	}
	cout<<"Generated String: "<<str1<<endl;
}